### ISSUE TITLE
Number of months for resub events is inaccurate

### ISSUE THREAD

**Issue by sg4e** [twitch4j/twitch4j#77]
### Prerequisites
* [ ] If this is a question/suggestion, I have already considered discuss it on [Discord Server](https://discord.gg/FQ5vgW3)
* [x] This issue specifically has something to do with Twitch4J
* [x] I have attempted to [look for similar issues](https://github.com/twitch4j/twitch4j/issues) already

### Description
Looked at the code and the Twitch API docs. msg-param-months is what Twitch4j parses for the SubscriptionEvent's getMonths() method. Here's what the Twitch API docs say about that parameter:
```
(Sent only on subgift, anonsubgift) The total number of months the user has subscribed. This is the same as msg-param-cumulative-months but sent for different types of user notices.
```
Here's msg-param-cumulative-months:
```
(Sent only on sub, resub) The total number of months the user has subscribed. This is the same as msg-param-months but sent for different types of user notices.
```
...which is bizarre. Twitch felt it was important to use distinct parameters for # of months subscribed dependent on whether the current resub was from a giftsub or regular sub.
Sub streak is a completely separate parameter: `msg-param-streak-months`.
Looking at the Twitch4j documentation, I can't determine what is specified to be returned from `SubscriptionEvent.getMonths()`; it's merely specified as months.
In my opinion `getMonths()` should return cumulative months subscribed and new methods should be introduced to query sub streak (which the subscriber can choose to share or hide).
Currently `getMonths()` is accurate only if it's a 1st-time sub or a giftsub.

### Steps to Reproduce

Listen for subscription events in a channel and compare sub-months to what `SubscriptionEvent.getMonths()` reports.

---

**Expected behavior:** 
Should return correct sub-month count.

**Actual behavior:** 
Only accurate for giftsubs.

**Stacktrace (if applicable):** 

**Version affected:**

**Comment by sg4e** [comment]
I'd like to work on this and submit a PR.

**Comment by PhilippHeuer** [comment]
I wasn't aware of that change or of all those new types for the usernotice messages you mentioned - thanks for the PR :)

**Comment by sg4e** [comment]
Fixed by f079e679f0b770914a4d5683b19130ad91120fd5
